### ISSUE TITLE
Package preinstallation in interactive mode

### ISSUE THREAD

**Issue by mallniya** [freebsd/poudriere#534]
I wonder if it's possible to preinstall packages in jail or run some bootstrap for the jail while using `poudriere bulk -i`?
This can be useful as some debugging occurs in the jail, so it's great to have your shell/texed/lint or such in an environment.
If no please consider this as feature request.

**Comment by bdrewery** [comment]
Yes we could add a `pre-interactive` hook (https://github.com/freebsd/poudriere/wiki/hooks). I'll add this for 3.2.

**Comment by rootwyrm** [comment]
Giving this a bump because my workflow has exposed a related issue. I basically do layered sets; `production` has the ports that apply to everything, then I have 'sub-sets' like `mail`, `www`, etc. which contain either additional ports or changed options (now that `priority` is working in pkgng.)

You can already guess where this is going. And obviously, it's not something that can be done in hooks easily if at all. (Would have to determine the full package set to build, calculate the dependency tree, compare dependencies against the set list, etc.) Being able to point poudriere to `production` and tell it "attempt pkg install for a dependency first from here *before* doing a build" would save quite literally hundreds of hours just in llvm* alone.
